### ISSUE TITLE
Cannot use ALLOW_MEMORY_GROWTH when execute the standalone wasm.

### ISSUE THREAD

**Issue by morgen52** [emscripten-core/emscripten#18197]
Hello everyone, last week I tried to use my wasm runtime (WasmEdge) to execute the standalone wasm compiled from c code by emcc. The main content of my compile command is ``emcc -O3 -s WASM=1 -s ALLOW_MEMORY_GROWTH=1 atax.c -DPOLYBENCH_TIME -DNDEBUG -o atax_time.wasm -s STANDALONE_WASM''

But I got this error. 

<img width="1058" alt="201567071-6dc695af-54e1-46a7-ba4f-b1b14f257882" src="https://user-images.githubusercontent.com/48060316/201583667-053a6c67-ae18-41a8-ab5b-d7b731efae2a.png">

I guess it's because the "env" and "emscripten_notify_memory_growth" should be loaded in js. So how can I set allow_memory_growth when I want to execute standalone wasm in my wasm runtime?

PS: Following is my output of "emcc -v"
<img width="1059" alt="201565783-1e9209da-36b3-42e1-ba7c-b8b92cfb03da" src="https://user-images.githubusercontent.com/48060316/201583702-417050b1-4087-493c-9fc0-1a98662c81d3.png">

Thanks for any help!

**Comment by kripken** [comment]
That import could be a no-op in a wasm VM, in principle. The problem though is that even in standalone wasm we can emit a JS file, and if the wasm is run with that JS, then we *do* need that import to function (which tells the JS runtime to refresh the memory views).

I think we could at least solve this for cases where there is no JS at all, by making a stub for `emscripten_notify_memory_growth` that does nothing. That could a file alongside the existing `standalone.c`, something like `standalone_no_js.c`, perhaps? And it is included only when emcc is not emitting any JS at all.

Alternatively, perhaps we could work around this on the JS side, but the workaround would require checking for the need to refresh memory views constantly on the JS boundary, which is why the more efficient import was added.

**Comment by sbc100** [comment]
Yes this is an interesting tradeoff.   I guess ideally we would have some kind of new mode such as "REALLY_STANDALONE", and I think we have discussed this in the past.  In fact, IIRC I've proposed such a change in that past.   

On the other hand the existing STANDALONE_WASM flag does say: `STANDALONE_WASM indicates that we want to emit a wasm file that can run  without JavaScript.`.   So maybe we should just skip this call completely in this mode?  (i.e. add a stub for standalone.c)

**Comment by kripken** [comment]
Yes, we could skip it - that would make the wasm run in wasm VMs, but not run with the JS, though. We have to pick one or the other. But maybe it's fine to pick the VM, as that is the more important use case.

If we agree it's more important to support VMs, perhaps we can make the JS way work with the slow checks for refreshing the views, that is, if the JS matters less, maybe it's fine it's slower, but we can at least make it work and not be broken.
